src/java/org/jivesoftware/openfire/plugin/util/cache/ClusterClassLoader.java

```diff
@@ -61,10 +65,24 @@ public class ClusterClassLoader extends ClassLoader {
         .setDynamic(false)
         .setPlugin(HazelcastPlugin.PLUGIN_NAME)
         .build();
+
+    private static final SystemProperty<Duration> CLASS_CACHE_DURATION = SystemProperty.Builder.ofType(Duration.class)
+        .setKey("hazelcast.cache.class.duration")
+        .setChronoUnit(ChronoUnit.MILLIS)
+        .setDefaultValue(Duration.ofSeconds(5))
+        .setDynamic(false)
+        .setPlugin(HazelcastPlugin.PLUGIN_NAME)
+        .build();
+
     private static final PluginManager pluginManager = XMPPServer.getInstance().getPluginManager();
 
     private final PluginClassLoader hazelcastClassloader;
 
+    // TODO: unload classes provided by plugins that are unloaded. Until then, keep the expiry very short, as to evict classes soon after unloading a plugin.
```

Comment:
My normal comment; if there's a TODO, at least raise a ticket to cover it off later, and reference that ticket in the comment.

Reply:
I don't know that a clusterable cache is a good idea. Even if a class can be serialized, I doubt that it would ever work on any other node than the node it was serialized on. Classpaths can be different, leading to subtly different classes being loaded (e.g. in a mixed-version cluster).

Reply:
Oh, I agree. We have `CacheFactory.createLocalCache()` that I was planning to use for this. The implementation will be `org.jivesoftware.util.cache.DefaultLocalCacheStrategy`, which has the benefit of implementing Openfire's `Cache` interface (and thus show up in the cache summary pages, etc).